### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by yoggys**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,55 +1,51 @@
-const client = require('discord-rich-presence')('1155259326554980473');
+const client = require("discord-rich-presence")("1155259326554980473");
 const CSGOGSI = require("./index");
-let gsi = new CSGOGSI({
-    port: 3000,
-    authToken: []
-});
```
Comment (yoggys) [inline app.js:6]: passing default values is not necessary

File: app.js [hunk 2]
```diff
@@ -1,55 +1,51 @@
...
 
-let presence = {
-    details: 'Waiting for game to start...',
-    largeImageKey: 'csgo',
-    largeImageText: 'CS:GO',
-    startTimestamp: Date.now(),
-}
```
Comment (yoggys) [inline app.js:13]: default value is not necessary as we call resetGamePhaseTimer

File: app.js [hunk 3]
```diff
@@ -1,55 +1,51 @@
...
-    }, 3000);
 };
 
-gsi.on("roundPhase", (phase) => {
-    resetGamePhaseTimer()
+client.on("ready", () => {
+    resetGameStateTimer();
+    new CSGOGSI()
+        .on("gameUpdate", (map) => setMapInfo(map))
+        .on("gameState", () => resetGameStateTimer());
```
Comment (yoggys) [inline app.js:50]: GSI start after client is ready

File: index.js [hunk 4]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
+            console.info(
+                `[@] CSGO2 GSI server listening on ${addr.address}:${addr.port}`
+            );
+        });
+    }
+
+    createServer() {
+        return http.createServer((req, res) => {
             if (req.method !== "POST") {
-                res.writeHead(404, { "Content-Type": "text/plain" });
```
Comment (yoggys) [inline index.js:15]: 405 is proper http code for this case

File: index.js [hunk 5]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
             req.on("end", () => {
-                this.processJson(body);
-                return res.writeHead(200).end();
+                try {
+                    this.processJson(body);
+                    res.writeHead(200).end();
+                } catch (e) {
+                    console.error("Error processing JSON:", e);
+                    res.writeHead(500).end();
+                }
```
Comment (yoggys) [inline index.js:36]: proper use of try-catch

File: index.js [hunk 6]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
-            this.emit("gamePhase", data["map"]["phase"]); //warmup etc
-            this.emit("gameRounds", data["map"]["round"]);
-            this.emit("gameCTscore", data["map"]["team_ct"]["score"]);
-            this.emit("gameTscore", data["map"]["team_t"]["score"]);
-            this.gameMap = data["map"]["name"];
-            this.gameMode = data["map"]["mode"];
-            this.gamePhase = data["map"]["phase"];
-            this.gameRounds = data["map"]["round"];
-            this.ctScore = data["map"]["team_ct"]["score"];
-            this.tScore = data["map"]["team_t"]["score"];
```
Comment (yoggys) [inline index.js:67]: as we emit values, this is not neccessary

File: index.js [hunk 7]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
-
-
     process(data) {
-        if (data["map"]) {
-            this.emit("gameMap", data["map"]["name"]);
-            this.emit("gameMode", data["map"]["mode"])
-            this.emit("gamePhase", data["map"]["phase"]); //warmup etc
-            this.emit("gameRounds", data["map"]["round"]);
-            this.emit("gameCTscore", data["map"]["team_ct"]["score"]);
-            this.emit("gameTscore", data["map"]["team_t"]["score"]);
```
Comment (yoggys) [inline index.js:61]: single emit is enough for this

File: index.js [hunk 8]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
-            this.emit("roundPhase", data["round"]["phase"]);
-            switch (data["round"]["phase"]) {
-                case "live":
-                    break;
-                case "freezetime":
-                    break;
-                case "over":
-                    this.emit("roundWinTeam", data["round"]["win_team"]);
-                    break;
-            }
```
Comment (yoggys) [inline index.js:88]: unused

File: index.js [hunk 9]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
-            this.tScore = data["map"]["team_t"]["score"];
-        }
-
-        if (data["round_wins"]) {
-            this.emit("roundWins", data["round_wins"]);
-        }
-
-        if (data["player"]) {
-            this.emit("player", data["player"]);
-        }
```
Comment (yoggys) [inline index.js:76]: unused

File: index.js [hunk 10]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
+                } catch (e) {
+                    console.error("Error processing JSON:", e);
+                    res.writeHead(500).end();
+                }
             });
         });
-
-        this.bombTime = 40;
-        this.isBombPlanted = false;
-        this.bombTimer = null;
```
Comment (yoggys) [inline index.js:32]: unused

File: app.js [hunk 11]
```diff
@@ -1,55 +1,51 @@
...
+    }, 3000);
+};
+
+const setMapInfo = (map) => {
+    const gameMap = data["map"]["name"];
+    const gameMode = data["map"]["mode"];
+    const gamePhase = data["map"]["phase"];
+    const gameRounds = data["map"]["round"] || 1;
+    const ctScore = data["map"]["team_ct"]["score"] || 0;
+    const tScore = data["map"]["team_t"]["score"] || 0;
```
Comment (yoggys) [inline app.js:26]: moved from CSI -> single listener for changes

File: index.js [hunk 12]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
-
-        if (data["round_wins"]) {
-            this.emit("roundWins", data["round_wins"]);
-        }
-
-        if (data["player"]) {
-            this.emit("player", data["player"]);
-        }
-
-        if (data["round"]) {
```
Comment (yoggys) [inline index.js:78]: if !round -> If the round does not exist, we are not in the game

File: index.js [hunk 13]
```diff
@@ -4,91 +4,63 @@ const EventEmitter = require("events");
...
-                    break;
-                case "over":
-                    this.emit("roundWinTeam", data["round"]["win_team"]);
-                    break;
-            }
-
+        if (!data["round"]) {
+            this.emit("gameState");
+        } else if (data["map"]) {
+            this.emit("gameUpdate", data["map"]);
```
Comment (yoggys) [inline index.js:61]: if we are not in the game, we change the game state, if we are in the game, we change the game info

**Review by itsxales — COMMENTED** [review]
Spotted some bugs, I will fix'em soon :)

**Review by itsxales — APPROVED** [review]
Approve.

**Comment by itsxales** [comment]
Thank you @yoggys for help!!!

**Pull request merged**